samples/api/oit_depth_peeling/oit_depth_peeling.cpp

```diff
@@ -0,0 +1,558 @@
+	camera.set_rotation({0.0f, 0.0f, 0.0f});
+	camera.set_perspective(60.0f, static_cast<float>(width) / static_cast<float>(height), 16.0f, 0.1f);
+
+	load_assets();
+	create_samplers();
+	create_constant_buffers();
+	create_descriptors();
+	create_sized_objects(width, height);
+	create_pipelines();
+
+	update_scene_constants();
+	update_descriptors();
+	build_command_buffers();
+
+	prepared = true;
+	return true;
+}
+
+bool OITDepthPeeling::resize(const uint32_t width, const uint32_t height)
+{
+	destroy_sized_objects();
+	create_sized_objects(width, height);
+	update_descriptors();
+	ApiVulkanSample::resize(width, height);
+	return true;
+}
+
+void OITDepthPeeling::render(float delta_time)
+{
+	if (!prepared)
+	{
+		return;
+	}
+
+	back_layer_index = std::max(back_layer_index, front_layer_index);
+
+	ApiVulkanSample::prepare_frame();
+	submit_info.commandBufferCount = 1;
+	submit_info.pCommandBuffers    = &draw_cmd_buffers[current_buffer];
+	VK_CHECK(vkQueueSubmit(queue, 1, &submit_info, VK_NULL_HANDLE));
+	ApiVulkanSample::submit_frame();
+
+	if (camera_auto_rotation)
+	{
+		camera.rotate({delta_time * 5.0f, delta_time * 5.0f, 0.0f});
+	}
+	update_scene_constants();
+}
+
+void OITDepthPeeling::request_gpu_features(vkb::PhysicalDevice &gpu)
+{
+	if (gpu.get_features().samplerAnisotropy)
+	{
+		gpu.get_mutable_requested_features().samplerAnisotropy = VK_TRUE;
+	}
+	else
+	{
+		throw std::runtime_error("This sample requires support for anisotropic sampling");
+	}
+}
+
+void OITDepthPeeling::on_update_ui_overlay(vkb::Drawer &drawer)
+{
+	drawer.checkbox("Camera auto-rotation", &camera_auto_rotation);
+	drawer.slider_float("Background grayscale", &background_grayscale, kBackgroundGrayscaleMin, kBackgroundGrayscaleMax);
+	drawer.slider_float("Object opacity", &object_alpha, kObjectAlphaMin, kObjectAlphaMax);
+
+	drawer.slider_int("Front layer index", &front_layer_index, 0, kLayerMaxCount - 1);
+	drawer.slider_int("Back layer index", &back_layer_index, 0, kLayerMaxCount - 1);
+}
+
+void OITDepthPeeling::build_command_buffers()
+{
+	VkCommandBufferBeginInfo command_buffer_begin_info = vkb::initializers::command_buffer_begin_info();
+
+	VkClearValue clear_values[2];
+	clear_values[0].color        = {{0.0f, 0.0f, 0.0f, 0.0f}};
+	clear_values[1].depthStencil = {0.0f, 0};
+
+	VkRenderPassBeginInfo render_pass_begin_info    = vkb::initializers::render_pass_begin_info();
+	render_pass_begin_info.renderArea.offset.x      = 0;
+	render_pass_begin_info.renderArea.offset.y      = 0;
+	render_pass_begin_info.renderArea.extent.width  = width;
+	render_pass_begin_info.renderArea.extent.height = height;
+	render_pass_begin_info.clearValueCount          = 2;
+	render_pass_begin_info.pClearValues             = clear_values;
+
+	for (int32_t i = 0; i < draw_cmd_buffers.size(); ++i)
+	{
+		VK_CHECK(vkBeginCommandBuffer(draw_cmd_buffers[i], &command_buffer_begin_info));
+		{
+			// Gather passes
+			for (uint32_t l = 0; l < kLayerMaxCount; ++l)
```

Comment:
I cannot skip the front layers completely, as I need their depth information to extract the subsequent layers. Technically, I could skip rendering any color attachment for layers before **front_layer_index**, but that would require additional changes for what is more of a "debug" visualization feature.

I did the simple change of skipping any layer after **back_layer_index** though, since nor their depth nor color are not used in any way.

Is this OK with you?